engine/table/src/main/java/io/deephaven/engine/table/impl/replay/ReplayTable.java

```diff
@@ -91,12 +94,19 @@ public void run() {
         if (done) {
             return;
         }
-        final RowSet added = advanceToCurrentTime();
-        if (added.isNonempty()) {
-            getRowSet().writableCast().insert(added);
-            notifyListeners(added, RowSetFactory.empty(), RowSetFactory.empty());
-        } else {
-            added.close();
+        try {
+            final RowSet added = advanceToCurrentTime();
+            if (added.isNonempty()) {
+                getRowSet().writableCast().insert(added);
+                notifyListeners(added, RowSetFactory.empty(), RowSetFactory.empty());
+            } else {
+                added.close();
+            }
+        } catch (final Exception err) {
+            updateSourceRegistrar.removeSource(this);
+
+            // propagate the error to our listeners
+            notifyListenersOnError(err, null);
```

Comment:
Why don't we have a UPL node? Maybe we should be using an `io.deephaven.engine.table.impl.InstrumentedUpdateSource`.